### DISCUSSION TITLE
How to remove an item from store?

### DISCUSSION THREAD

**Discussion by weijarz** [solidjs/solid#1211] [Q&A] [answered]
<https://playground.solidjs.com/?hash=2005854203&version=1.4.1>

```ts
  const [state, setState] = createStore<Record<string, string>>({
    firstName: "John",
    lastName: "Miller",
  });

  setState('firstName', undefined) // It actually looks like it works, but has type errors
```

In version 1.4.1 there is no type error to write this way:
```ts
setState('firstName', () => undefined)
```
But there has type error in 1.5.x

**Comment by martinpengellyphillips** [comment, marked as answer]
`setState('firstName', undefined)` is correct and does remove the item from the store.

For the typing error you can either:

a) Explicitly mark it as okay `setState('firstName', undefined!)`
b) Include `undefined` in your store typing: `createStore<Record<string, string | undefined>>(...`

**Reply by weijarz** [reply]
Thank you! I learned `undefined!` this useful new typescript trick.
